src/cdk/tree/navigator.ts

```diff
@@ -0,0 +1,311 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FocusableOption} from '@angular/cdk/a11y';
+import {Directionality} from '@angular/cdk/bidi';
+import {
+  DOWN_ARROW,
+  END,
+  ENTER,
+  HOME,
+  LEFT_ARROW,
+  NUMPAD_MULTIPLY,
+  RIGHT_ARROW,
+  TAB,
+  UP_ARROW
+} from '@angular/cdk/keycodes';
+import {Directive, Optional} from '@angular/core';
+import {Subject} from 'rxjs';
+import {TreeControl} from './control/tree-control';
+
+
+/**
+ * Keyboard navigator interface.
+ */
+// TODO(tinayuangao): Move to a common place
+export interface CdkNavigator<T> {
```

Comment:
Could the name be more specific than "Navigator"? The name alone would make me think it has something to do with routing.

Reply:
Changed to `CdkTreeKeyboardInteraction` and removed the interface since it's hard to make it work for the table. Any suggestion for the name?  `CdkTreeKeyManager`?

Reply:
`CdkTreeKeyManager` sounds good and aligns with the other key managers we have